test/websocket/issue-2844.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const { test } = require('node:test')
+const { once } = require('node:events')
+const { WebSocketServer } = require('ws')
+const { WebSocket } = require('../..')
+const { tspl } = require('@matteo.collina/tspl')
+
+test('The server must reply with at least one subprotocol the client sends', async (t) => {
+  const { completed, deepStrictEqual, fail } = tspl(t, { plan: 2 })
+
+  const wss = new WebSocketServer({
+    handleProtocols: (protocols) => {
+      deepStrictEqual(protocols, new Set(['msgpack', 'json']))
+
+      return protocols.values().next().value
+    },
+    port: 0
+  })
+
+  wss.on('connection', (ws) => {
+    ws.on('error', fail)
+    ws.send('something')
+  })
+
+  await once(wss, 'listening')
+
+  const ws = new WebSocket(`ws://localhost:${wss.address().port}`, {
+    protocols: ['msgpack', 'json']
```

Comment:
The second constructor parameter must be subprotocol string or array of subprotocol strings, not object

Reply:
we allow objects too https://github.com/nodejs/undici/blob/main/docs/docs/api/WebSocket.md#new-websocketurl-protocol